Add syslog support for slog handlers

rosstimothy opened the pull request:
Updates the syslog writer creation to allow both logrus and slog to output to syslog.

zmb3 commented on lib/config/configuration.go:
```diff
@@ -681,18 +681,22 @@ func applyLogConfig(loggerConfig Log, cfg *servicecfg.Config) error {
-		// TODO(tross): add slog support for syslog.
-		hook, err := utils.CreateSyslogHook()
+		w = os.Stderr
+		sw, err := utils.NewSysLogWriter()
```
```suggestion
		sw, err := utils.NewSyslogWriter()
```

Nit: it's Syslog, not SysLog.

rosstimothy committed:
Update slog handlers to support configured extra_fields (#36372)

In addition to configuring output format users can configure the
fields which are included in log messages. Prior to this our slog
handlers didn't support this feature and always output the default
fields. Now, both the slog and logrus output is identical no matter
the extra_fields configuration as verified by TestExtraFields. It
was also found that due to the way the logurs json formatter was
configuring itself in CheckAndSetDefaults the timestamp was always
included in the final output even if it was not included in the
configured extra_fields.
